WordPress/gutenberg
# Remove all wp.api Usage from the editor package

This PR tries to remove all wp.api usage from the "editor" package.

**Notes**

 - Splits effects file for sanity and also to make it easier to use `async/await`
 - Adds a `resolveSelector` utility function that waits for the resolution of the selector before returning a promise with its value.
 - Replaces `wp.api.getPostTypeRoute` with `select( 'core' ).getPostType( getCurrentPostType( state ) ).rest_base`

**Testing instructions**

 - Ensure shared blocks work
 - Ensure taxonomy selectors work
 - Ensure saving posts work
